config/routes.rb

```diff
@@ -21,6 +21,14 @@
       end
     end
 
+    %i[ violence ].each do |step|
+      resource step, only: %i[ show update ], controller: :risks, step: step do
+        match '/',
+          action: :update_and_redirect_to_profile, via: %i[ put patch ],
+          constraints: -> (r) { r.params['commit'] =~ /Save and view profile/ }
```

Comment:
😐 I shouldn't have let this approach go into the healthcare feature, if the button content changes in the view it will break this - the values need to be pulled from a single source. I just ran a test on master and altered the value being checked in the routing constraint, to my surprise the test suite passed successfully. I know the same can be said for the adding move destinations but if that is changed a spec fails. When another additional multiple was added the values being checked in the constraint should have been extracted to a single source.